src/main/java/com/deviceinsight/kafka/health/KafkaConsumingHealthIndicator.java

```diff
@@ -95,8 +95,10 @@ void shutdown() {
 
 	private void setConsumerGroup(Map<String, Object> kafkaConsumerProperties) {
 		try {
-			kafkaConsumerProperties.putIfAbsent(ConsumerConfig.GROUP_ID_CONFIG,
-					"health-check-" + InetAddress.getLocalHost().getHostAddress());
+			String groupId = (String) kafkaConsumerProperties.getOrDefault(ConsumerConfig.GROUP_ID_CONFIG,
+					UUID.randomUUID().toString());
+			kafkaConsumerProperties.put(ConsumerConfig.GROUP_ID_CONFIG,
+					groupId + "-health-check-" + InetAddress.getLocalHost().getHostAddress());
```

Comment:
I would argue that `health-check-` should be a prefix, so health check consumer groups are easier to find when looking at the consumer groups in Kafka.